### PR TITLE
Ask va api updating categories mock data

### PR DESCRIPTION
**Pull request by Khoa-V-Nguyen**
## Summary

This pull request introduces several key changes to improve content handling for categories, topics, and subtopics within the AskVA API:

1. **Update categories, topics, and subtopics with CRM API fields**
   - Integrated new fields introduced by the CRM API into the categories, topics, and subtopics data structures.
   - Ensures alignment with the latest CRM API changes, improving data consistency.

2. **Refactor: Consolidate categories, topics, and subtopics into `AskVAApi::Contents`**
   - Merged the separate entities (categories, topics, and subtopics) into a unified `AskVAApi::Contents` module.
   - This refactor simplifies data retrieval and reduces duplication, enhancing maintainability and readability.

3. **Add `contents` action to `StaticDataController`**
   - Added a new `contents` action in `StaticDataController` to handle the consolidated content retrieval through a single endpoint.
   - Streamlines access to categories, topics, and subtopics, improving the API's usability and efficiency for consumers.
 
4. **Refactor**: Streamline argument passing to `BaseRetriever` with `**args`
   - Replaced explicit passing of `user_mock_data` and other arguments with `**args` in all retriever classes.
   - This approach simplifies code, reduces redundancy, and improves flexibility for future argument changes.
   - Ensures arguments are passed directly to `BaseRetriever` without modification, enhancing maintainability and readability.

--- 

This message captures the essence of the changes you’ve made across the classes.

## Related issue(s)

- https://github.com/department-of-veterans-affairs/ask-va/issues/1184

## Testing done

- [x] *New code is covered by unit tests*
- [ ] * Use `Postman` to access the `endpoint` for further testing.

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [x]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x]  Feature/bug has a monitor built into Datadog (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected

### CODE REVIEW

File: modules/ask_va_api/app/lib/ask_va_api/contents/entity.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+                  :parent_id,
+                  :description,
+                  :requires_authentication,
+                  :allow_attachments,
+                  :rank_order,
+                  :display_name,
+                  :topic_type,
+                  :contact_preferences
+
+      def initialize(info)
```
Comment (LindseySaari) [inline modules/ask_va_api/app/lib/ask_va_api/contents/entity.rb:17]: Are you ok with nil values here if any of these info attributes don't exist?

File: modules/ask_va_api/config/locales/static_data.json [hunk 2]
```diff
@@ -1,1974 +1,2350 @@
...
-          "ParentId": "0d2b8586-e764-eb11-bb23-000d3a579c3f",
-          "Description": null,
-          "RequiresAuthentication": false,
-          "AllowAttachments": true,
-          "RankOrder": 0,
-          "DisplayName": "Special benefits application status",
-          "TopicType": "SubTopic",
-          "ContactPreferences": []
-      }
+    {
```
Comment (LindseySaari) [inline modules/ask_va_api/config/locales/static_data.json:3]: Can we just store this in the db instead? Does it have to be a jSON file? I see it's probably static, but does it ever change?

File: modules/ask_va_api/app/lib/ask_va_api/contents/entity.rb (hunk 1 above)
Comment (Khoa-V-Nguyen) [reply modules/ask_va_api/app/lib/ask_va_api/contents/entity.rb:17]: Yes, it's not a hard requirement.

File: modules/ask_va_api/config/locales/static_data.json (hunk 2 above)
Comment (Khoa-V-Nguyen) [reply modules/ask_va_api/config/locales/static_data.json:3]: This is a mock data file for development. My api doesn't use a db, I'm relying on CRM API as my db.

**Review by ediiotero — APPROVED** [review]
LGTM

**Comment by rmtolmach** [comment]
@Khoa-V-Nguyen I see you have a lot of unit tests, but how do you plan to test this in dev or staging?

**Comment by Khoa-V-Nguyen** [comment]
> @Khoa-V-Nguyen I see you have a lot of unit tests, but how do you plan to test this in dev or staging?

I'm planning on to use `Postman` to retrieve the data and see if everything comes out correctly.

File: modules/ask_va_api/config/locales/static_data.json [hunk 3]
```diff
@@ -1,1974 +1,2350 @@
...
+      "ParentId": "f0ba9562-e864-eb11-bb23-000d3a579c44",
+      "Description": null,
+      "RequiresAuthentication": false,
+      "AllowAttachments": false,
+      "RankOrder": 0,
+      "DisplayName": "Reporting a broken link on VA.gov",
+      "TopicType": "SubTopic",
+      "ContactPreferences": [
+
+      ]
```
Comment (rmtolmach) [inline modules/ask_va_api/config/locales/static_data.json:15]: What are the changes in this file? Is it this change compared to `"ContactPreferences": []` being on one line?

Comment (Khoa-V-Nguyen) [reply modules/ask_va_api/config/locales/static_data.json:15]: There are additional data and some of the name got changed. This file is only here for FE to access in local. The normal flow will access the cache data.

**Comment by rmtolmach** [comment]
> > @Khoa-V-Nguyen I see you have a lot of unit tests, but how do you plan to test this in dev or staging?
> 
> I'm planning on to use `Postman` to retrieve the data and see if everything comes out correctly.

Awesome. Would you mind adding that to the **Testing** section in the description?

**Review by rmtolmach — APPROVED** [review]

**Pull request merged**